src/tango_sdp_subarray/SDPSubarray/SDPSubarray.py

```diff
@@ -84,24 +85,55 @@ def delete_device(self):
     # Attributes methods
     # ------------------
 
-    def write_obsState(self, obs_state):
-        """Set obsState."""
+    def write_obsState(self, obs_state: ObsState):
+        """Set the obsState attribute.
+
+        :param obs_state: An observation state enum value.
+        """
         self.obs_state = obs_state
 
-    def read_obsState(self):
-        """Get obsState."""
+    def read_obsState(self) -> ObsState:
+        """Get the obsState attribute.
+
+        :returns: The current obsState attribute value.
+        """
         return self.obs_state
 
-    def write_adminMode(self, admin_mode):
-        """Set adminMode."""
+    def write_adminMode(self, admin_mode: AdminMode):
```

Comment:
Do we need to define what adminMode is, or is this inherited from the Tango base classes?

Reply:
that is a good question! While we can inherit from adminMode, currently the way enums are handled in the base classes is not that great in that it uses an older style rather than what is described here https://pytango.readthedocs.io/en/stable/data_types.html#devenum-pythonic-usage. 

What we really need to do is fix the base classes but as that was a LOT more work, for the moment i've redefined some state attributes in a way that allows us to use the newer style but keeps the enum values consistent with that in the base class.

Reply:
I was going to change this back to use the inherited versions if the NCRA team complained but as i've not heard anything yet i'm assuming they are happy with or have not noticed this.

Reply:
I think that's all fair enough, and we can leave this as is.